semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -563,6 +563,9 @@ and m_expr a b =
   Trace_matching.(if on then print_expr_pair a b);
   match (a.G.e, b.G.e) with
   (* the order of the matches matters! take care! *)
+  (* alias: match on the expr inside *)
+  | G.Alias (_alias, a1), _ -> m_expr a1 b
```

Comment:
Alternatively perhaps this one should only match when both are aliases

Reply:
Whether it uses the dataflow engine or not, you could require `symbolic_propogation` to be enabled. Then, if it's not enabled, both sides would have to be an `Alias` for it to match.